src/compiler/checker.ts

```diff
@@ -22051,6 +22051,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     // For example, `foo-${number}` is related to `foo-${string}` even though number isn't related to string.
                     instantiateType(source, reportUnreliableMapper);
                 }
+                if (isTemplateLiteralTypeWithSinglePlaceholderAndNoExtraTexts(target) && isTypeAssignableTo(source, stringType) && isTypeAssignableTo(target.types[0], stringType)) {

```

Comment:
Why do we need to check that `source` is assignable to `string`?

Reply:
My reasoning is that we can relate `T` and `${T}` only if both are actually strings. Template literal placeholders might be non-strings so without that we'd accidentally allow matching a string with a number and stuff like that.

Reply:
Ok, so we have source `${S}` and target `T`, and I can see why we check that `S` <: `string`. But in that case, isn't `${S}` the same as `S`?